addons/website_sale/static/src/scss/product_tile.scss

```diff
@@ -411,20 +411,40 @@
         }
     }
 
-    &:where(.o_wsale_products_opt_wishlist_inline.o_wsale_products_opt_actions_inline) {
+    &:where(.o_wsale_products_opt_actions_inline:not(.o_wsale_products_opt_wishlist_fixed):not(.o_wsale_products_opt_wishlist_fixed_onhover)) {
         .o_wsale_product_btn {
-            @container (max-width: 270px) {
+            @container (max-width: 200px) {
                 --o-wsale-comparison-btn-placeholder-display: none;
             }
+        }
 
-            &:where(:has(.o_add_wishlist):has(.o_add_compare)) {
-                @include media-breakpoint-up(lg) {
-                    @container (max-width: 250px) {
-                        --o-wsale-card-btn-submit-label-display: none;
-                        --o-wsale-card-btn-submit-flex-grow: 0;
-                        --o-wsale-card-btn-submit-aspect-ratio: 1;
-                        --o-wsale-card-btn-submit-margin: 0 auto 0 0;
-                    }
+        @mixin hide-submit-button-label {
+            --o-wsale-card-btn-submit-label-display: none;
+            --o-wsale-card-btn-submit-flex-grow: 0;
+            --o-wsale-card-btn-submit-aspect-ratio: 1;
+            --o-wsale-card-btn-submit-margin: 0 auto 0 0;
+        }
+
+        &:where(.o_wsale_products_opt_has_comparison) {
+            .o_wsale_product_btn:where(:has(.o_add_compare)) {
+                @container (max-width: 245px) {
+                    @include hide-submit-button-label;
+                }
+            }
+        }
+
+        &:where(.o_wsale_products_opt_has_wishlist) {
+            .o_wsale_product_btn:where(:has(.o_add_wishlist)) {
+                @container (max-width: 245px) {
+                    @include hide-submit-button-label;
+                }
+            }
+        }
+
+        &.o_wsale_products_opt_has_wishlist.o_wsale_products_opt_has_comparison {
+            .o_wsale_product_btn:where(:has(.o_add_wishlist):has(.o_add_compare)) {
+                @container (max-width: 250px) {
+                    @include hide-submit-button-label;
```

Comment:
On mobile there's this small issue, it's because the comparison button is always hidden. 🥲 
<img width="744" height="932" alt="Capture d’écran 2025-08-25 à 08 31 11" src="https://github.com/user-attachments/assets/79f3093f-69a3-4be9-aec3-c708dfe10452" />